src/main/java/com/bobocode/bibernate/session/SessionFactoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.bobocode.bibernate.session;
+
+import com.bobocode.bibernate.PooledDataSourceProvider;
+import com.bobocode.bibernate.configuration.Dialect;
+import com.bobocode.bibernate.configuration.PersistenceUnitProperties;
+import com.bobocode.bibernate.exception.BibernateException;
+import com.bobocode.parser.PropertyParser;
+
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.bobocode.parser.YamlPropertyParser.toPersistenceUnitProperties;
+
+
+public class SessionFactoryImpl implements SessionFactory {
+
+    private final DataSource dataSource;
+    private final Dialect dialect;
+    private final PersistenceUnitProperties properties;
+
+    public SessionFactoryImpl(PropertyParser parser, String persistenceUnit) {
+        Objects.requireNonNull(persistenceUnit, "persistence unit name must not be null");
```

Comment:
I think this check is redundant, ve validate persistanceUnitName in `PersistenceUnitProperties` class, and in `YamlParser` as well.

P.S yes I know this is my code, but I don`t  commit in your PR without comments, so can I ask you to fix it?

Reply:
Sure. I will remove it.